src/python/zquantum/core/utils.py

```diff
@@ -339,7 +340,7 @@ def load_value_estimate(file):
     return ValueEstimate.from_dict(data)
 
 
-def save_value_estimate(value_estimate, filename):
+def save_value_estimate(value_estimate: ValueEstimate, filename: DumpTarget):
```

Comment:
I recommend renaming `filename` to just `file`, as it can be either a file or a path.